### PR TITLE
DataGrid.Pdf - Implement styles for backgroundColor, textColor, borderColor

### PR DESCRIPTION
**Pull request by KuznetsovVN**

### CODE REVIEW

File: js/exporter/jspdf/v3/export_data_grid_3.js [hunk 1]
```diff
@@ -24,7 +24,8 @@ function exportDataGrid(doc, dataGrid, options) {
     return new Promise((resolve) => {
         dataProvider.ready().done(() => {
             // TODO: pass rowOptions: { headerStyles: { backgroundColor }, groupStyles: {...}, totalStyles: {...} }
-            const rowsInfo = generateRowsInfo(dataProvider, dataGrid);
+            const rowOptions = options.rowOptions || {};
```
Comment (IgnatovDan) [inline js/exporter/jspdf/v3/export_data_grid_3.js]: 'const rowOptions' - excessive variable?
Can we remove it?

File: js/exporter/jspdf/v3/rows_generator.js [hunk 2]
```diff
@@ -43,7 +43,7 @@ function generateRowsInfo(dataProvider, dataGrid, headerStyles) {
                 columns,
                 rowType,
                 colCount: columns.length,
-                backgroundColor: (rowType === 'header') ? headerStyles?.backgroundColor : undefined
+                backgroundColor: (rowType === 'header') ? rowOptions?.headerStyles?.backgroundColor : undefined
```
Comment (IgnatovDan) [inline js/exporter/jspdf/v3/rows_generator.js]: A more complex object will be passed:

`headerStyles?.backgroundColor -> rowOptions?.headerStyles?.backgroundColor 
`

Can we pass the used 'headerBackgroundColor' value only?

**Review by IgnatovDan — CHANGES_REQUESTED** [review]

File: js/exporter/jspdf/v3/export_data_grid_3.js (hunk 1 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/v3/export_data_grid_3.js]: Ok, I rolled back this excessive variable

File: js/exporter/jspdf/v3/rows_generator.js (hunk 2 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/v3/rows_generator.js]: Done

File: js/exporter/jspdf/v3/draw_utils.js [hunk 3]
```diff
@@ -63,18 +75,31 @@ function drawBorders(doc, rect, drawLeftBorder = true, drawRightBorder = true, d
...
 
-function setCurrentFont(doc, cell, styles) {
-    const font = isDefined(cell.font) ? extend({}, styles.font, cell.font) : styles.font;
+function setTextStyles(doc, { textColor, font }, docStyles) {
+    const _textColor = isDefined(textColor) ? textColor : docStyles.textColor;
+    if(_textColor !== doc.getTextColor()) {
+        doc.setTextColor(_textColor);
+    }
+
+    const _font = isDefined(font) ? extend({}, docStyles.font, font) : docStyles.font;
```
Comment (novsstation) [inline js/exporter/jspdf/v3/draw_utils.js]: In general variable with "_" menas private property.
Maybe it's better to rename it to cellFont or currentFont?
Same as for _textColor

File: js/exporter/jspdf/v3/draw_utils.js [hunk 4]
```diff
@@ -4,46 +4,58 @@ import { extend } from '../../../core/utils/extend';
...
     }
 }
 
-function drawCellsLines(doc, cellsArray) {
-    cellsArray.forEach(cell => {
-        // TODO: doc.setDrawColor(borderColor); // cell.borderColor OR docStyles.borderColor
-        drawBorders(doc, cell._rect, cell.drawLeftBorder, cell.drawRightBorder, cell.drawTopBorder, cell.drawBottomBorder);
-    });
+function drawCellsLines(doc, cellsArray, docStyles) {
+    cellsArray
```
Comment (novsstation) [inline js/exporter/jspdf/v3/draw_utils.js:30]: del. LGTM

File: js/exporter/jspdf/v3/draw_utils.js (hunk 3 above)
Comment (KuznetsovVN) [reply js/exporter/jspdf/v3/draw_utils.js]: Ok, good idea. Let's do this renaming.

**Review by IgnatovDan — APPROVED** [review]

**Review by novsstation — APPROVED** [review]

**Pull request merged**